sklearn/model_selection/_split.py

```diff
@@ -440,13 +440,16 @@ class GroupKFold(_BaseKFold):
     The same group will not appear in two different folds (the number of
     distinct groups has to be at least equal to the number of folds).
 
-    The folds are approximately balanced in the sense that the number of
-    distinct groups is approximately the same in each fold.
-
     Parameters
     ----------
     n_splits : int, default=3
         Number of folds. Must be at least 2.
+    method: string, default='balance'
+        One of 'balance', 'stratify', 'shuffle'.
+        By default, try to equalize the sizes of the resulting folds.
+        If 'stratify', sort groups according to ``y`` variable and distribute
```

Comment:
It only sorts groups according to the `y` variable of the first sample for that group, not, say the mean or the mode or whatever. I think the stratify case is most useful when there is a many-to-one relationship between group and target. And I wonder if we should enforce that relationship (throw an error if there are many y values for some group) just to make this logic explicable, straightforward, and invariant to reordering the samples.